Document an official Docker Compose setup

Toshaksha opened the pull request:
<!-- 
Thanks for opening a pull request to Prefect! 
If this is your first contribution, please make sure to review our contribution guidelines: https://docs.prefect.io/contribute/index
-->

### Summary
Add a comprehensive guide for running a self-hosted Prefect server environment using Docker Compose.  
This new documentation (`docker-compose.mdx`) includes:
- Docker Compose configuration for PostgreSQL, Redis, Prefect server, services, and worker
- Instructions to run and stop the entire stack
- Notes on configuration and service dependencies
- Troubleshooting tips for common issues

### Why?
Providing this official guide helps users quickly deploy Prefect server locally or in production-like environments with Docker, improving ease of use and reducing setup errors.

closes #18494

### Checklist
<!-- These boxes may be checked after opening the pull request. -->

- [x] This pull request references any related issue by including "closes #18494"
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

zzstoatzz reviewed (changes requested):
I think this is a good start @Toshaksha - but I believe we're missing several things here

- the redis messaging broker is not configured here. if we're going to include redis, we should use it ([example](https://github.com/zzstoatzz/prefect-pack/blob/main/examples/run_a_prefect_server/docker_compose/multi-server/compose.yaml#L55-L59))
- the server will migrate on start by default, but like the example linked above, I think the migration should happen in a one-time command that the server `depends_on` and then we should set the `MIGRATE_ON_START=false` on the `prefect-server` service

i will review again later

zzstoatzz commented on docs/v3/how-to-guides/self-hosted/docker-compose.mdx:
```diff
@@ -0,0 +1,145 @@
+      PREFECT_REDIS_MESSAGING_HOST: redis
+      PREFECT_REDIS_MESSAGING_PORT: 6379
+      PREFECT_REDIS_MESSAGING_DB: 0
+      MIGRATE_ON_START: "false"
```
this would be `PREFECT_SERVER_MIGRATE_ON_START` but on second thought, i think we can omit this for now - sorry for the whiplash
```suggestion
```

zzstoatzz commented on docs/v3/how-to-guides/self-hosted/docker-compose.mdx:
```diff
@@ -0,0 +1,145 @@
+    volumes:
+      - redis_data:/data
+
+  prefect-migrate:
+    image: prefecthq/prefect:3-latest
+    environment:
+      PREFECT_API_DATABASE_CONNECTION_URL: postgresql+asyncpg://prefect:prefect@postgres:5432/prefect
+    depends_on:
+      postgres:
+        condition: service_healthy
+    command: >
+      bash -c "prefect server database upgrade || true"
```
after discussing with the team, i think we can omit this for now - my bad

zzstoatzz commented on docs/v3/how-to-guides/self-hosted/docker-compose.mdx:
```diff
@@ -0,0 +1,145 @@
+      - redis
+      - prefect-server
+    environment:
+      PREFECT_API_URL: http://localhost:4200/api
```
should this refer to `prefect-server` instead of localhost?

zzstoatzz reviewed (commented):
thanks for the quick turnaround! a couple more comments

Toshaksha commented:
Thanks for the feedback, @zzstoatzz!
I removed the migrate service for clarity, simplified the Redis configuration, and added healthchecks for Postgres and Redis. The docs have been updated accordingly.

Please let me know if you have any further guidance before the final review.
